modules/creating-a-machine-pool-cli.adoc

```diff
@@ -28,6 +28,9 @@ $ rosa create machinepool --cluster=<cluster-name> \
                           --taints=<key>=<value>:<effect>,<key>=<value>:<effect> \ <5>
                           --use-spot-instances \ <6>
                           --spot-max-price=0.5 <7>
+ifdef::openshift-rosa[]
+                          --worker-disk-size \ <8>
```

Comment:
Let's clarify worker disk size only work for non-hosted cluster.

Reply:
There should be a value set for it. ` --worker-disk-size ` like ` --worker-disk-size  200GiB`

Reply:
And create machinepool uses flag `--disk-size` rather than ` --worker-disk-size`
lixue@Xue-Lis-MacBook-Pro rosa % rosa create machinepool -h
Add a machine pool to the cluster.

Usage:
  rosa create machinepool [flags]

Aliases:
  machinepool, machinepools, machine-pool, machine-pools

Examples:
  # Interactively add a machine pool to a cluster named "mycluster"
  rosa create machinepool --cluster=mycluster --interactive

  # Add a machine pool mp-1 with 3 replicas of m5.xlarge to a cluster
  rosa create machinepool --cluster=mycluster --name=mp-1 --replicas=3 --instance-type=m5.xlarge

  # Add a machine pool mp-1 with autoscaling enabled and 3 to 6 replicas of m5.xlarge to a cluster
  rosa create machinepool --cluster=mycluster --name=mp-1 --enable-autoscaling \
	--min-replicas=3 --max-replicas=6 --instance-type=m5.xlarge

  # Add a machine pool with labels to a cluster
  rosa create machinepool -c mycluster --name=mp-1 --replicas=2 --instance-type=r5.2xlarge --labels=foo=bar,bar=baz,

  # Add a machine pool with spot instances to a cluster
  rosa create machinepool -c mycluster --name=mp-1 --replicas=2 --instance-type=r5.2xlarge --use-spot-instances \
    --spot-max-price=0.5

Flags:
      --autorepair                 Select auto-repair behaviour for a machinepool in a hosted cluster. (default true)
      --availability-zone string   Select availability zone to create a single AZ machine pool for a multi-AZ cluster
  -c, --cluster string             Name or ID of the cluster.
      --disk-size string           Root disk size with a suffix like GiB or TiB
      --enable-autoscaling         Enable autoscaling for the machine pool.
  -h, --help                       help for machinepool
      --instance-type string       Instance type that should be used. (default "m5.xlarge")
  -i, --interactive                Enable interactive mode.
      --labels string              Labels for machine pool. Format should be a comma-separated list of 'key=value'. This list will overwrite any modifications made to Node labels on an ongoing basis.
      --max-replicas int           Maximum number of machines for the machine pool.
      --min-replicas int           Minimum number of machines for the machine pool.
      --multi-availability-zone    Create a multi-AZ machine pool for a multi-AZ cluster (default true)
      --name string                Name for the machine pool (required).
  -o, --output string              Output format. Allowed formats are [json yaml]
      --replicas int               Count of machines for the machine pool (required when autoscaling is disabled).
      --spot-max-price string      Max price for spot instance. If empty use the on-demand price. (default "on-demand")
      --subnet string              Select subnet to create a single AZ machine pool for BYOVPC cluster
      --taints string              Taints for machine pool. Format should be a comma-separated list of 'key=value:ScheduleType'. This list will overwrite any modifications made to Node taints on an ongoing basis.
      --tuning-configs string      Name of the tuning configs to be applied to the machine pool. Format should be a comma-separated list. Tuning config must already exist. This list will overwrite any modifications made to node tuning configs on an ongoing basis.
      --use-spot-instances         Use spot instances for the machine pool.
      --version string             Version of OpenShift that will be used to install a machine pool for a hosted cluster, for example "4.12.4"

Global Flags:
      --color string     Surround certain characters with escape sequences to display them in color on the terminal. Allowed options are [auto never always] (default "auto")
      --debug            Enable debug mode.
      --profile string   Use a specific AWS profile from your credential file.
      --region string    Use a specific AWS region, overriding the AWS_REGION environment variable.
  -y, --yes              Automatically answer yes to confirm operation.
lixue@Xue-Lis-MacBook-Pro rosa %

Reply:
this feature content is for ROSA Classic only. There is a separate feature/effort for ROSA HCP.